### PR TITLE
Lantern Bugs Fix

### PR DESCRIPTION
**Pull request by emichaud998**
Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

Changes in this PR:
Addressed the following bugs:
- _There are 9 Epic Endpoints on the "SMART Response" page that are all the same endpoint but just have different formatting things going on._  While we could not really remove all the URLs with slight typo variations, made it so every endpoint that gets stored in the fhir_endpoints table is preceded by `https://` and ends with a backslash, which will remove any duplicate URL entries that differ either by how the URL scheme is formatted (http, https, HTTP, HTTPS, etc.) or by whether they end with a slash or not, making the URLs a little more consistent. 
- _There are a few epic endpoints that are returning a failure for the "generalMimeType" validation, but it looks like they should be passing because their value is "application/json+fhir,application/fhir+json" (but their FHIR version is unknown? not sure what's going on there)_ Any endpoint that once had a successful response will have the MIME types saved in the fhir_endpoints_info table for that endpoint, and even if they are not responding anymore the MIME types will be persisted for that endpoint in the DB. Therefore, the generalMimeType validation now will not set the Actual value equal to the valid MIME types that an endpoint has stored from previous successful requests when the endpoint is no longer responding and the FHIR version is thus unknown. 
- _Can't download the CSV file (but looks like JSON is working)_ Fixed this bug so the CSV file can now be downloaded
- _Did the location page ever show the endpoint that an organization mapped to? I thought it did but I could be misremembering_ The location page did not show the endpoint that an organization is mapped to, but it has been added

**Submitter:**
- [x] Administrative tasks are complete.
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: [Lantern 427 JIRA Ticket](https://oncprojectracking.healthit.gov/support/browse/LANTERN-427)
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
- [x] Code review has been performed.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

- [ ] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [x] Code review has been performed.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
- [ ] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

- [ ] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [ ] Code review has been performed.
  - Code accomplishes the tasks purpose.
- [ ] Tests are complete.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge
is complete.

Create a new branch `<branch name>_gomodupdate`. Run `make update_mods` to update the `go.mod`
and `go.sum` files to point to master. Create a PR. Require one sanity check reviewer.

### CODE REVIEW

File: endpointmanager/pkg/fhirendpointquerier/populatefhirendpoints.go [hunk 1]
```diff
@@ -75,6 +76,9 @@ func formatToFHIREndpt(endpoint *fetcher.EndpointEntry) (*endpointmanager.FHIREn
 		uri = uri + "/"
 	}
 
+	splitEndpoint := strings.Split(uri, "://")
+	uri = "https://" + splitEndpoint[len(splitEndpoint)-1]
```
Comment (bmath10) [inline endpointmanager/pkg/fhirendpointquerier/populatefhirendpoints.go:80]: Based on the conversation with Jeff in Slack, I think what we should do here is instead of always tacking on `https://`, to instead use toLower() if the http[s] exists, to at least catch the HTTPS vs https issue, and tack it on if it doesn't exist. This won't catch the issue where you have the same URL where the only difference is http/https, but I think it's probably better than messing with an endpoint where it only works for http and we accidentally break it.

File: shinydashboard/lantern/modules/endpointsmodule.R [hunk 2]
```diff
@@ -117,7 +117,7 @@ endpointsmodule <- function(
   # Create the format for the csv
   csv_format <- reactive({
     res <- selected_fhir_endpoints() %>%
-      select(-updated, -label, -status, -availability) %>%
+      select(-label, -status, -availability) %>%
```
Comment (bmath10) [inline shinydashboard/lantern/modules/endpointsmodule.R:120]: Another bug that I just noticed is that the Field Descriptions CSV now doesn't align with the table, since the mime types is now supported formats and the valid capability statement field was added. Could you update that CSV?

File: endpointmanager/pkg/fhirendpointquerier/populatefhirendpoints.go (hunk 1 above)
Comment (emichaud998) [reply endpointmanager/pkg/fhirendpointquerier/populatefhirendpoints.go:80]: I agree, I think this makes more sense!

**Comment by bmath10** [comment]
The fix for the URLs all look good, except that there was one URL from the problematic URL set that didn't have the http in front of it? It was `epicproxy.et0927.epichosted.com/FHIRProxy/api/FHIR/DSTU2/` with the NPI ID `{1275158958}`

**Review by bmath10 — APPROVED** [review]
